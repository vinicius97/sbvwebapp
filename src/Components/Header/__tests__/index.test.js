import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import { Link } from 'react-router-dom'

import { Header } from '../'
import Logotipo from  '../images/sambatech.svg'

configure({adapter: new Adapter()})

describe('<Header />', () => {

  const wrapper = shallow(<Header homePage={'/homepage'}/>)

  it('Comparar snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Validar exibição de link Home', () => {
    const linkHome = wrapper.find(Link)

    expect(linkHome).toHaveLength(1)
    expect(linkHome.prop('to')).toEqual('/homepage')

    wrapper.setProps({homePage: null})
    expect(wrapper.find(Link)).toHaveLength(0)
  })

  it('Validar exibição do logotipo', () => {
    const logotipo = wrapper.find('.header--container__logo')

    expect(logotipo).toHaveLength(1)
    expect(logotipo.prop('src')).toEqual(Logotipo)
  })

})