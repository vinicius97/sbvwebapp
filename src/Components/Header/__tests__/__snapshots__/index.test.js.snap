// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Comparar snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    homePage="/homepage"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="header--container__back"
          replace={false}
          to="/homepage"
>
          Home
</Link>,
        <img
          alt="SambaTech"
          className="header--container__logo"
          src="sambatech.svg"
/>,
      ],
      "className": "header--container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Home",
          "className": "header--container__back",
          "replace": false,
          "to": "/homepage",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "SambaTech",
          "className": "header--container__logo",
          "src": "sambatech.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="header--container__back"
            replace={false}
            to="/homepage"
>
            Home
</Link>,
          <img
            alt="SambaTech"
            className="header--container__logo"
            src="sambatech.svg"
/>,
        ],
        "className": "header--container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "className": "header--container__back",
            "replace": false,
            "to": "/homepage",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "SambaTech",
            "className": "header--container__logo",
            "src": "sambatech.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
